src/Components/Web/src/Forms/InputSelect.cs

```diff
@@ -24,16 +27,51 @@ public class InputSelect<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTy
         /// </summary>
         [DisallowNull] public ElementReference? Element { get; protected set; }
 
+        /// <summary>
+        /// Gets or sets the current value of the input as a string array.
+        /// </summary>
+        protected string?[]? CurrentValueAsStringArray
+        {
+            get
+            {
+                if (CurrentValue is not Array array)
+                {
+                    return null;
+                }
+                
+                return array
+                    .Cast<object?>()
+                    .Select(item => item?.ToString())
+                    .ToArray();
+            }
+            set
+            {
+                CurrentValue = BindConverter.TryConvertTo<TValue>(value, CultureInfo.InvariantCulture, out var result)
+                    ? result
+                    : default;
+            }
+        }
+
         /// <inheritdoc />
         protected override void BuildRenderTree(RenderTreeBuilder builder)
         {
             builder.OpenElement(0, "select");
             builder.AddMultipleAttributes(1, AdditionalAttributes);
             builder.AddAttribute(2, "class", CssClass);
-            builder.AddAttribute(3, "value", BindConverter.FormatValue(CurrentValueAsString));
-            builder.AddAttribute(4, "onchange", EventCallback.Factory.CreateBinder<string?>(this, __value => CurrentValueAsString = __value, CurrentValueAsString));
-            builder.AddElementReferenceCapture(5, __selectReference => Element = __selectReference);
-            builder.AddContent(6, ChildContent);
+
+            if (AdditionalAttributes?.ContainsKey("multiple") ?? false)
+            {
+                builder.AddAttribute(3, "value", BindConverter.FormatValue(CurrentValue)?.ToString());
+                builder.AddAttribute(4, "onchange", EventCallback.Factory.CreateBinder(this, __value => CurrentValueAsStringArray = __value, CurrentValueAsStringArray));
```

Comment:
I think we can make this slightly more efficient. It's a bit weird, but `EventCallback.Factory.CreateBinder<T>` doesn't actually use the *existingValue* parameter value at runtime - it's only there at compile-time to coerce the compiler into inferring the correct generic type.

Since in this case it's quite expensive to call `get_CurrentValueAsStringArray`, we could replace that parameter value with `default(string)`.

It might even be worth removing the getter for `CurrentValueAsStringArray` entirely if there's no anticipated use case for it.

Reply:
Ah, good catch. I had noticed that at one point but didn't think to take advantage of it here.